astropy/coordinates/baseframe.py

```diff
@@ -156,6 +157,67 @@ def __new__(mcls, name, bases, members):
                 'Could not find all expected BaseCoordinateFrame class '
                 'attributes.  Are you mis-using FrameMeta?')
 
+        # Unless overridden, velocity name defaults are:
+        #   * pm_{lon}_cos{lat} for SphericalCosLatDifferential
+        #   * radial_velocity for any d_distance, including RadialDifferential
+        #   * v_{x,y,z} for d_{x,y,z} for CartesianDifferential
+        # where {lon} and {lat} are the names of the angular components
+        if repr_info is None:
+            repr_info = {}
+
+        for cls_or_name in repr_info.keys():
+            if isinstance(cls_or_name, str):
+                # TODO: this provides a layer of backwards compatibility in
+                # case the key is a string, but now we want explicit classes.
+                cls = _get_repr_cls(cls_or_name)
+                repr_info[cls] = repr_info.pop(cls_or_name)
+
+        # The default spherical names are 'lon' and 'lat'
+        repr_info.setdefault(r.SphericalRepresentation,
+                             [RepresentationMapping('lon', 'lon'),
+                              RepresentationMapping('lat', 'lat')])
+
+        sph_component_map = {m.reprname: m.framename
+                             for m in repr_info[r.SphericalRepresentation]}
+
+        repr_info.setdefault(r.SphericalCosLatDifferential, [
+            RepresentationMapping(
+                'd_lon_coslat',
+                'pm_{lon}_cos{lat}'.format(**sph_component_map),
+                u.mas/u.yr),
+            RepresentationMapping('d_lat',
+                                  'pm_{lat}'.format(**sph_component_map),
+                                  u.mas/u.yr),
+            RepresentationMapping('d_distance', 'radial_velocity',
+                                  u.km/u.s)
+        ])
+
+        repr_info.setdefault(r.SphericalDifferential, [
+            RepresentationMapping('d_lon',
+                                  'pm_{lon}'.format(**sph_component_map),
+                                  u.mas/u.yr),
+            RepresentationMapping('d_lat',
+                                  'pm_{lat}'.format(**sph_component_map),
+                                  u.mas/u.yr),
+            RepresentationMapping('d_distance', 'radial_velocity',
+                                  u.km/u.s)
+        ])
+
+        repr_info.setdefault(r.CartesianDifferential, [
+            RepresentationMapping('d_x', 'v_x', u.km/u.s),
+            RepresentationMapping('d_y', 'v_y', u.km/u.s),
+            RepresentationMapping('d_z', 'v_z', u.km/u.s)])
+
+        # Unit* classes should follow the same naming conventions
+        repr_info.setdefault(r.UnitSphericalRepresentation,
+                             repr_info[r.SphericalRepresentation])
+
+        repr_info.setdefault(r.UnitSphericalCosLatDifferential,
```

Comment:
Maybe idiotic sense of correctness, but let's not give a radial mapping if it cannot exist, i.e., write `repr_info[r.SphericalCosLatDifferential][:2]`

Reply:
Users are allowed to define their frame classes with any order of `RepresentationMapping`'s, so I don't think the slicing is enough. I could do something like, e.g.,
```python
r_info = [x for x in repr_info[r.SphericalRepresentation] 
          if x.reprname != 'distance']
repr_info.setdefault(r.UnitSphericalRepresentation, r_info)
```
for each case. Thoughts?

Reply:
Maybe in that case it isn't worth it.

Reply:
I'm ok with leaving this alone for now instead of worrying about making the logic air-tight.  Might leave a TODO that checking this might be worth adding, though?

Reply:
OK comment added